test/extensions/stats_sinks/hystrix/hystrix_test.cc

```diff
@@ -0,0 +1,370 @@
+namespace StatSinks {
+namespace Hystrix {
+
+class HystrixSinkTest : public testing::Test {
+public:
+  HystrixSinkTest() { sink_.reset(new HystrixSink(server_, 10)); }
+
+  absl::string_view getStreamField(absl::string_view dataMessage, absl::string_view key) {
+    absl::string_view::size_type key_pos = dataMessage.find(key);
+    EXPECT_NE(absl::string_view::npos, key_pos);
+    absl::string_view trimDataBeforeKey = dataMessage.substr(key_pos);
+    key_pos = trimDataBeforeKey.find(" ");
+    EXPECT_NE(absl::string_view::npos, key_pos);
+    absl::string_view trimDataAfterValue = trimDataBeforeKey.substr(key_pos + 1);
+    key_pos = trimDataAfterValue.find(",");
+    EXPECT_NE(absl::string_view::npos, key_pos);
+    absl::string_view actual = trimDataAfterValue.substr(0, key_pos);
+    return actual;
+  }
+
+  Buffer::OwnedImpl createClusterAndCallbacks() {
+
+    // set cluster
+    // cluster1_.info_->name_ = cluster1_name_;
+    cluster_map_.emplace("test_cluster1", cluster1_);
+    ON_CALL(server_, clusterManager()).WillByDefault(ReturnRef(cluster_manager_));
+    ON_CALL(cluster_manager_, clusters()).WillByDefault(Return(cluster_map_));
+    ON_CALL(cluster1_, info()).WillByDefault(Return(cluster1_info_ptr_));
+
+    ON_CALL(*cluster1_info_, name()).WillByDefault(testing::ReturnRefOfCopy(cluster1_name_));
+    ON_CALL(*cluster1_info_, statsScope()).WillByDefault(ReturnRef(cluster1_stats_scope_));
+
+    // set gauge value.
+    membership_total_gauge_.name_ = "membership_total";
+    ON_CALL(cluster1_stats_scope_, gauge("membership_total"))
+        .WillByDefault(ReturnRef(membership_total_gauge_));
+    ON_CALL(membership_total_gauge_, value()).WillByDefault(Return(5));
+
+    // attach counters
+    success_counter_.name_ = "upstream_rq_2xx";
+    ON_CALL(cluster1_stats_scope_, counter("upstream_rq_2xx"))
+        .WillByDefault(ReturnRef(success_counter_));
+    error_5xx_counter_.name_ = "upstream_rq_5xx";
+    ON_CALL(cluster1_stats_scope_, counter("upstream_rq_5xx"))
+        .WillByDefault(ReturnRef(error_5xx_counter_));
+    retry_5xx_counter_.name_ = "retry.upstream_rq_5xx";
+    ON_CALL(cluster1_stats_scope_, counter("retry.upstream_rq_5xx"))
+        .WillByDefault(ReturnRef(retry_5xx_counter_));
+    error_4xx_counter_.name_ = "upstream_rq_4xx";
+    ON_CALL(cluster1_stats_scope_, counter("upstream_rq_4xx"))
+        .WillByDefault(ReturnRef(error_4xx_counter_));
+    retry_4xx_counter_.name_ = "retry.upstream_rq_4xx";
+    ON_CALL(cluster1_stats_scope_, counter("tetry.upstream_rq_4xx"))
+        .WillByDefault(ReturnRef(retry_4xx_counter_));
+    ON_CALL(retry_5xx_counter_, value()).WillByDefault(Return(0));
+    ON_CALL(error_4xx_counter_, value()).WillByDefault(Return(0));
+    ON_CALL(retry_5xx_counter_, value()).WillByDefault(Return(0));
+
+    // set callbacks to send data to buffer
+    Buffer::OwnedImpl buffer;
+    auto encode_callback = [&buffer](Buffer::Instance& data, bool) {
+      buffer.add(
```

Comment:
I'd put the comment above the call to `buffer.add(data);` -- I think it'll format more nicely that way.